packages/wrangler/src/config/config-helpers.ts

```diff
@@ -1,35 +1,127 @@
-import path from "path";
+import fs from "node:fs";
+import path from "node:path";
 import { findUpSync } from "find-up";
+import dedent from "ts-dedent";
+import { UserError } from "../errors";
+import { logger } from "../logger";
+import { parseJSONC, readFileSync } from "../parse";
+
+export type ResolveConfigPathOptions = {
+	useRedirect?: boolean;
+};
 
 /**
  * Resolve the path to the configuration file, given the `config` and `script` optional command line arguments.
  * `config` takes precedence, then `script`, then we just use the cwd.
  */
-export function resolveWranglerConfigPath({
-	config,
-	script,
-}: {
-	config?: string;
-	script?: string;
-}): string | undefined {
+export function resolveWranglerConfigPath(
+	{
+		config,
+		script,
+	}: {
+		config?: string;
+		script?: string;
+	},
+	options: { useRedirect?: boolean }
+): string | undefined {
 	if (config !== undefined) {
 		return config;
 	}
 
 	const leafPath = script !== undefined ? path.dirname(script) : process.cwd();
-	return findWranglerConfig(leafPath);
+
+	return findWranglerConfig(leafPath, options);
 }
 
 /**
  * Find the wrangler config file by searching up the file-system
  * from the current working directory.
  */
 export function findWranglerConfig(
-	referencePath: string = process.cwd()
+	referencePath: string = process.cwd(),
+	{ useRedirect = false } = {}
 ): string | undefined {
-	return (
+	const userConfigPath =
 		findUpSync(`wrangler.json`, { cwd: referencePath }) ??
 		findUpSync(`wrangler.jsonc`, { cwd: referencePath }) ??
-		findUpSync(`wrangler.toml`, { cwd: referencePath })
-	);
+		findUpSync(`wrangler.toml`, { cwd: referencePath });
+
+	return useRedirect
+		? findRedirectedWranglerConfig(referencePath, userConfigPath)
+		: userConfigPath;
+}
+
+/**
+ * Check whether there is a config file that indicates
+ * @param cwd
+ * @param userConfigPath
+ * @returns
+ */
+function findRedirectedWranglerConfig(
+	cwd: string,
+	userConfigPath: string | undefined
+) {
+	const PATH_TO_DEPLOY_CONFIG = ".wrangler/deploy/config.json";
+	const deployConfigPath = findUpSync(PATH_TO_DEPLOY_CONFIG, { cwd });
+	if (deployConfigPath === undefined) {
```

Comment:
what does this mean tho? Basically the code in the calling fn becomes:
```
configPath: useRedirect
			? userConfigPath
			: userConfigPath,
```
Does this mean we set `useRedirect` but forgot to provide a redirect config path? Won't returning userConfigPath without any warning result in an unexpected behaviour for users? Just thinking out loud

Reply:
after reading more of the code, I **think** this makes sense, because `useRedirect` serves as a a flag that indicates we should check whether there's a redirect config? For example our deploy commands, which can't know beforehand whether a redirect config is present, so we set `useRedirect` to `true` by default, which would ensure we always check for a redirect file. If no redirect file is found, then we proceed with the user config file. Did I get this right? And do I win a star if I did? 🌟 😋

Reply:
Yes, I guess we could rename `useRedirect` to something like `tryRedirect`, which might make it clearer that we are trying to use a redirect if one is available in a `.wrangler/deploy/config.json`?

Reply:
Renamed to `useRedirectIfAvailable`.